external-crates/move/crates/move-analyzer/editors/code/src/context.ts

```diff
@@ -177,25 +203,199 @@ export class Context {
      */
     registerOnDidChangeConfiguration(): void {
         vscode.workspace.onDidChangeConfiguration(async event => {
-            const changed = event.affectsConfiguration('move.lint') ||
-                            event.affectsConfiguration('move.inlay-hints.type');
-            if (changed) {
-                const newLintLevel = lint();
-                const newInlayHintsType = inlayHintsForType();
-                if (this.lintLevel !== newLintLevel ||
-                    this.inlayHintsType !== newInlayHintsType) {
-                    this.lintLevel = newLintLevel;
-                    this.inlayHintsType = newInlayHintsType;
-                    try {
-                        await this.stopClient();
+
+            const server_path_conf = MOVE_CONF_NAME.concat('.').concat(SERVER_PATH_OPT);
+            const sui_path_conf = MOVE_CONF_NAME.concat('.').concat(SUI_PATH_OPT);
+            const lint_conf = MOVE_CONF_NAME.concat('.').concat(LINT_OPT);
+            const type_hints_conf = MOVE_CONF_NAME.concat('.').concat(TYPE_HINTS_OPT);
+
+            const optionsChanged = event.affectsConfiguration(lint_conf) ||
+                event.affectsConfiguration(type_hints_conf);
+            const pathsChanged = event.affectsConfiguration(server_path_conf) ||
+                event.affectsConfiguration(sui_path_conf);
+
+            if (optionsChanged || pathsChanged) {
+                this.configuration = new Configuration();
+                log.info(`configuration: ${this.configuration.toString()}`);
+
+                this.lintLevel = this.configuration.lint;
+                this.inlayHintsType = this.configuration.inlayHintsForType;
+                try {
+                    await this.stopClient();
+                        if (pathsChanged) {
+                            await this.installServerBinary(this.extensionContext);
+                        }
                         await this.startClient();
-                    } catch (err) {
-                        // Handle error
-                        log.info(String(err));
-                    }
+                } catch (err) {
+                    // Handle error
+                    log.info(String(err));
                 }
             }
         });
 
     }
+
+    /**
+     * Installs language server binary in the default location if needed.
+     * On a happy path we just compare versions of installed and bundled
+     * binaries and don't do any actual file operations at all.
+     *
+     * The actual algorithm deciding on how this works as follows.
```

Comment:
There are many ways to slice it and I think this one makes sense but I am open to other suggestions